spyder/app/mainwindow.py

```diff
@@ -1954,6 +1955,10 @@ def show_toolbars(self):
         self._update_show_toolbars_action()
 
     # --- Other
+    def free_memory(self):
+        """Free memory after event."""
+        gc.collect()

```

Comment:
Let's also add a timer that calls this method and is executed every minute and every 5 minutes after this method is called.